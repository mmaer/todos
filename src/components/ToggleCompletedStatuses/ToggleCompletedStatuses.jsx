import React from "react";

import "./ToggleCompletedStatuses.scss";

const ToggleCompletedStatuses = ({ toggleCompletedStatuses, todos }) => {
  const toggle = ({ target }) => {
    toggleCompletedStatuses(target.checked);
  };

  const allTodosCompleted = todos.every(({ completed }) => completed);

  return (
    <>
      <input
        id="toggle-completed-statuses"
        className="toggle-completed-statuses"
        type="checkbox"
        onChange={toggle}
        checked={allTodosCompleted}
      />
      <label htmlFor="toggle-completed-statuses">Mark all as complete</label>
    </>
  );
};

export default ToggleCompletedStatuses;
