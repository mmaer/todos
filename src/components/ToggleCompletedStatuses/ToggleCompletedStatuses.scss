.toggle-completed-statuses {
  text-align: center;
  border: none; /* Mobile Safari */
  opacity: 0;
  position: absolute;

  & + label {
    width: 60px;
    height: 34px;
    font-size: 0;
    position: absolute;
    top: -52px;
    left: -13px;
    transform: rotate(90deg);

    &:before {
      content: "‚ùØ";
      font-size: 22px;
      color: #e6e6e6;
      padding: 10px 27px 10px 27px;
    }
  }

  &:checked + label:before {
    color: #737373;
  }
}
