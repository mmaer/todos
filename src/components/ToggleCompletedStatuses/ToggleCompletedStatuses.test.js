import React from "react";
import { shallow, mount } from "enzyme";

import ToggleCompletedStatuses from "./";

const TODOS = [{ completed: false }];

describe("ToggleCompletedStatuses component", () => {
  it("Should match snapshot", () => {
    const wrapper = shallow(
      <ToggleCompletedStatuses
        toggleCompletedStatuses={() => {}}
        todos={TODOS}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("Should display checked checkbox when all todos are completed", () => {
    const wrapper = shallow(
      <ToggleCompletedStatuses
        toggleCompletedStatuses={() => {}}
        todos={[{ completed: true }, { completed: true }]}
      />
    );
    const checkbox = wrapper.find("input");
    expect(checkbox.props().checked).toBeTruthy();
  });

  it("Should display not checked checkbox when not all todos are completed", () => {
    const wrapper = shallow(
      <ToggleCompletedStatuses
        toggleCompletedStatuses={() => {}}
        todos={[{ completed: false }, { completed: true }]}
      />
    );
    const checkbox = wrapper.find("input");
    expect(checkbox.props().checked).toBeFalsy();
  });

  it("Should call toggleCompletedStatuses function when checkbox is clicked", () => {
    const mockToggleCompletedStatusesFunc = jest.fn();
    const wrapper = mount(
      <ToggleCompletedStatuses
        toggleCompletedStatuses={mockToggleCompletedStatusesFunc}
        todos={[{ completed: false }, { completed: true }]}
      />
    );
    const checkbox = wrapper.find("input");
    checkbox.simulate("change", { target: { checked: true } });

    expect(mockToggleCompletedStatusesFunc.mock.calls.length).toBe(1);
  });
});
