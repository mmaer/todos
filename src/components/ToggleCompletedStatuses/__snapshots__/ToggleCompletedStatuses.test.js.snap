// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleCompletedStatuses component Should match snapshot 1`] = `
<Fragment>
  <input
    checked={false}
    className="toggle-completed-statuses"
    id="toggle-completed-statuses"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    htmlFor="toggle-completed-statuses"
  >
    Mark all as complete
  </label>
</Fragment>
`;
